ui/src/app/shared/components/dropdown/dropdown.tsx

```diff
@@ -0,0 +1,127 @@
+import * as classNames from 'classnames';
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import {BehaviorSubject, Observable, Subscription} from 'rxjs';
+
+export interface DropDownProps {
+    isMenu?: boolean;
+    anchor: React.ComponentType;
+    children: React.ReactNode | (() => React.ReactNode);
+    qeId?: string;
+}
+
+export interface DropDownState {
+    opened: boolean;
+    left: number;
+    top: number;
+}
+
+require('./dropdown.scss');
+
+const dropDownOpened = new BehaviorSubject<DropDown>(null);
+
+export class DropDown extends React.Component<DropDownProps, DropDownState> {
+    private el: HTMLDivElement;
+    private content: HTMLDivElement;
+    private subscriptions: Subscription[];
+
+    constructor(props: DropDownProps) {
+        super(props);
+        this.state = {opened: false, left: 0, top: 0};
+    }
+
+    public render() {
+        let children: React.ReactNode = null;
+        if (typeof this.props.children === 'function') {
+            if (this.state.opened) {
+                const fun = this.props.children as () => React.ReactNode;
+                children = fun();
+            }
+        } else {
+            children = this.props.children as React.ReactNode;
+        }
+
+        return (
+            <div className='argo-dropdown' ref={el => (this.el = el)}>
+                <div
+                    qe-id={this.props.qeId}
+                    className='argo-dropdown__anchor'
+                    onClick={event => {
+                        this.open();
+                        event.stopPropagation();
+                    }}>
+                    <this.props.anchor />
+                </div>
+                {ReactDOM.createPortal(
+                    <div
+                        className={classNames('argo-dropdown__content', {'opened': this.state.opened, 'is-menu': this.props.isMenu})}
+                        style={{top: this.state.top, left: this.state.left}}
+                        ref={el => (this.content = el)}>
+                        {children}
```

Comment:
This is where we display the "filter" part of the code. This function does not make any assumptions about the content of `children` (which comes from `props.children`): The code in this function is only concerned with displaying the drop down menu and has no code regarding the actual list to be filtered.